cirq/ops/three_qubit_gates_test.py

```diff
@@ -167,30 +180,62 @@ def test_decomposition_respects_locality(gate):
 
 def test_diagram():
     a, b, c, d = cirq.LineQubit.range(4)
+    diagonal_angles = [np.pi * i / 4 for i in range(0, 8)]
     circuit = cirq.Circuit.from_ops(
         cirq.TOFFOLI(a, b, c),
-        cirq.TOFFOLI(a, b, c)**0.5,
-        cirq.CCX(a, c, b),
-        cirq.CCZ(a, d, b),
-        cirq.CCZ(a, d, b)**0.5,
-        cirq.CSWAP(a, c, d),
-        cirq.FREDKIN(a, b, c)
-    )
-    cirq.testing.assert_has_diagram(circuit, """
-0: ───@───@───────@───@───@───────@───@───
-      │   │       │   │   │       │   │
-1: ───@───@───────X───@───@───────┼───×───
-      │   │       │   │   │       │   │
-2: ───X───X^0.5───@───┼───┼───────×───×───
+        cirq.TOFFOLI(a, b, c)**0.5, cirq.CCX(a, c, b), cirq.CCZ(a, d, b),
+        cirq.CCZ(a, d, b)**0.5, cirq.CSWAP(a, c, d), cirq.FREDKIN(a, b, c),
+        cirq.ThreeQubitDiagonalGate(diagonal_angles)(a, b, c))
+    cirq.testing.assert_has_diagram(
+        circuit, """
+0: ───@───@───────@───@───@───────@───@───diag───
+      │   │       │   │   │       │   │   │
+1: ───@───@───────X───@───@───────┼───×───#2─────
+      │   │       │   │   │       │   │   │
+2: ───X───X^0.5───@───┼───┼───────×───×───#3─────
                       │   │       │
-3: ───────────────────@───@^0.5───×───────
+3: ───────────────────@───@^0.5───×──────────────
 """)
-    cirq.testing.assert_has_diagram(circuit, """
-0: ---@---@-------@---@---@-------@------@------
-      |   |       |   |   |       |      |
-1: ---@---@-------X---@---@-------|------swap---
-      |   |       |   |   |       |      |
-2: ---X---X^0.5---@---|---|-------swap---swap---
+    cirq.testing.assert_has_diagram(circuit,
+                                    """
+0: ---@---@-------@---@---@-------@------@------diag---
```

Comment:
The diagonal gate needs to describe the phases it is using. This will make the diagram quite wide or perhaps quite tall, but it is needed. Look at what `TwoQubitMatrixGate` does for reference.

Reply:
I kept it flat so that the circuit is wide.